src/metabase/api/database.clj

```diff
@@ -1097,22 +1104,36 @@
 
 ;;; ------------------------------------- GET /api/database/:id/schema/:schema ---------------------------------------
 
-(defn- schema-tables-list [db-id schema]
-  (api/read-check Database db-id)
-  (api/check-403 (can-read-schema? db-id schema))
-  (filter mi/can-read? (t2/select Table
-                         :db_id           db-id
-                         :schema          schema
-                         :active          true
-                         ;; a non-nil value means Table is hidden -- see [[metabase.models.table/visibility-types]]
-                         :visibility_type nil
-                         {:order-by [[:display_name :asc]]})))
+(defn- schema-tables-list
+  ([db-id schema include_hidden]
+   (api/read-check Database db-id)
+   (api/check-403 (can-read-schema? db-id schema))
+   (filter mi/can-read? (if include_hidden
+                          (t2/select Table
+                            :db_id db-id
+                            :schema schema
+                            :active true
+                            {:order-by [[:display_name :asc]]})
+                          (t2/select Table
+                            :db_id db-id
+                            :schema schema
+                            :active true
+                            :visibility_type nil
+                            {:order-by [[:display_name :asc]]}))))
+  ([db-id schema]
+   (schema-tables-list db-id schema nil)))
 
 (api/defendpoint GET "/:id/schema/:schema"
```

Comment:
We also need to make the fix work when there is no schema (or an empty schema). There is an endpoint for that in the same file `GET "/:id/schema/"`